src/test/java/uk/gov/pay/products/persistence/dao/ProductDaoIT.java

```diff
@@ -231,4 +236,69 @@ public void findById_shouldReturnMetadata_whenItExistsForAPaymentLink() {
         assertThat(productMetadataMap.containsKey("key2"), is(true));
         assertThat(productMetadataMap.containsValue("value2"), is(true));
     }
+
+    @Test
+    public void findProductsAndUsage_shouldReturnAProductUsage_whenExistsWithTypeAdhoc() {
+        ProductEntity productEntity = ProductEntityFixture.aProductEntity()
+                .withExternalId(randomUuid())
+                .withType(ProductType.ADHOC)
+                .withGatewayAccountId(1)
+                .build();
+        ProductEntity secondProductEntity = ProductEntityFixture.aProductEntity()
+                .withExternalId(randomUuid())
+                .withType(ProductType.ADHOC)
+                .withGatewayAccountId(2)
+                .build();
+        ProductEntity ignoredProductEntity = ProductEntityFixture.aProductEntity()
+                .withExternalId(randomUuid())
+                .withType(ProductType.DEMO)
+                .withGatewayAccountId(1)
+                .build();
+
+        productEntity = productDao.merge(productEntity);
+        secondProductEntity = productDao.merge(secondProductEntity);
+        ignoredProductEntity = productDao.merge(ignoredProductEntity);
+
+        PaymentEntity payment = PaymentEntityFixture.aPaymentEntity()
+                .withExternalId(randomUuid())
+                .withStatus(PaymentStatus.CREATED)
+                .withProduct(productEntity)
+                .withReferenceNumber("MH2KJY5KPW")
+                .build();
+        PaymentEntity secondPayment = PaymentEntityFixture.aPaymentEntity()
+                .withExternalId(randomUuid())
+                .withStatus(PaymentStatus.CREATED)
+                .withProduct(productEntity)
+                .withReferenceNumber("MH2KJY5KPW")
+                .build();
+        PaymentEntity thirdPayment = PaymentEntityFixture.aPaymentEntity()
+                .withExternalId(randomUuid())
+                .withStatus(PaymentStatus.CREATED)
+                .withProduct(secondProductEntity)
+                .withReferenceNumber("MH2KJY5KPW")
+                .build();
+        PaymentEntity ignoredPayment = PaymentEntityFixture.aPaymentEntity()
+                .withExternalId(randomUuid())
+                .withStatus(PaymentStatus.CREATED)
+                .withProduct(ignoredProductEntity)
+                .withReferenceNumber("MH2KJY5KPW")
+                .build();
+
+        databaseHelper.addPayment(payment.toPayment(), 1);
+        databaseHelper.addPayment(secondPayment.toPayment(), 1);
+        databaseHelper.addPayment(thirdPayment.toPayment(), 1);
+        databaseHelper.addPayment(ignoredPayment.toPayment(), 1);
+
+        List<ProductUsageStat> usageStats = productDao.findProductsAndUsage(null);
```

Comment:
(minor) This test can possibly be split into two for keeping contexts different - one with gateway_account_id and another without